arrow/src/main/java/org/apache/iceberg/arrow/vectorized/NullabilityHolder.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.arrow.vectorized;
+
+public class NullabilityHolder {
+  private final boolean[] isNull;
+  private int numNulls;
+
+  public NullabilityHolder(int batchSize) {
+    this.isNull = new boolean[batchSize];
+  }
+
+  public void setNull(int idx) {
+    isNull[idx] = true;
+    numNulls++;
+  }
+
+  public boolean isNullAt(int idx) {
+    return isNull[idx];
```

Comment:
All of the uses of `isNullAt` (at least, in this PR) convert boolean to 0 or 1 using a ternary operator. Should we store `isNull` as `byte[]` instead? Would that improve or hurt performance? Maybe there are uses elsewhere that make boolean the right choice here.